bpf/bpf_lxc.c

```diff
@@ -1524,35 +1669,30 @@ ipv4_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label, enum ct_status
 	    struct ipv4_ct_tuple *tuple_out, __u16 *proxy_port,
 	    bool from_host __maybe_unused)
 {
-	struct ct_state ct_state_on_stack, *ct_state = &ct_state_on_stack;
-	struct ipv4_ct_tuple tuple_on_stack, *tuple = &tuple_on_stack;
-	struct ct_state ct_state_new = {};
+	struct ct_state *ct_state, ct_state_new = {};
+	struct ipv4_ct_tuple *tuple;
 	void *data, *data_end;
 	struct iphdr *ip4;
 	bool skip_ingress_proxy = false;
 	bool is_untracked_fragment = false;
+	struct ct_buffer4 *ct_buffer;
 	int ret, verdict = 0, l4_off;
-	bool has_l4_header = false;
+	__u32 monitor = 0, zero = 0;
 	enum trace_reason reason;
-	__u32 monitor = 0;
 	__be32 orig_sip;
 	__u8 policy_match_type = POLICY_MATCH_NONE;
 	__u8 audited = 0;
 
 	if (!revalidate_data(ctx, &data, &data_end, &ip4))
 		return DROP_INVALID;
-	has_l4_header = ipv4_has_l4_header(ip4);
 
 	policy_clear_mark(ctx);
-	tuple->nexthdr = ip4->protocol;
 
 	/* If packet is coming from the ingress proxy we have to skip
 	 * redirection to the ingress proxy as we would loop forever.
 	 */
 	skip_ingress_proxy = tc_index_skip_ingress_proxy(ctx);
 
-	tuple->daddr = ip4->daddr;
-	tuple->saddr = ip4->saddr;
 	orig_sip = ip4->saddr;
 
 	l4_off = ETH_HLEN + ipv4_hdrlen(ip4);
```

Comment:
Keep things in sync with `ipv6_policy()`, and move `l4_off` down into the only branch where it's used?